gobblin-runtime/src/main/java/org/apache/gobblin/runtime/app/ServiceBasedAppLauncher.java

```diff
@@ -163,9 +170,17 @@ public void run() {
     });
 
     LOG.info("Starting the Gobblin application and all its associated Services");
+    for (Service service : this.services) {
+      LOG.info("Starting service " + service.getClass().getSimpleName());
+    }
 
     // Start the application
-    this.serviceManager.startAsync().awaitHealthy();
+    try {
+      this.serviceManager.startAsync().awaitHealthy(startupTimeout.getSeconds(), TimeUnit.SECONDS);
```

Comment:
One of the services, the FSConfigurationManager, has some main thread interrupting behavior that causes the start call to last indefinitely. 

This prevents users from being able to call the shutdown method because this method is synchronized. I'm adding a configuration for the timeout to be configurable, with the default being indefinitely waiting

Reply:
In this case, `startAsync` is not really working as expected when using the FSConfigurationManager. The actual work being done in cluster is done via the `FSConfigurationManager` when that class takes that config and produces a source out of the `.conf` file.

The `awaitHealthy` call here waits indefinitely and suspends this current thread in the current implementation. It does not impede our ability to fail-fast because often just hangs if there's an issue. If anything, we can fail faster because a configurable timeout allows us to unblock this thread and trigger the `shutdown` when the `Workflow` finishes (successfully or unsuccessfully)